app/src/main/java/com/example/taskyy/di/AuthModule.kt

```diff
@@ -33,6 +33,12 @@ object AuthModule {
         return AuthRepositoryImpl(userDao, api)
     }
 
+    @Provides
+    @Singleton
+    fun provideNavController(@ApplicationContext context: Context): NavController {
+        return NavController(context = context)
+    }
```

Comment:
I don't think this is the proper way to create a NavController for Compose. I would always prefer `rememberNavController` since that's aware of the current composable and will be properly destroyed when not needed anymore